etcdserver/api/v2http/client_auth.go

```diff
@@ -90,8 +90,7 @@ func hasKeyPrefixAccess(sec auth.Store, r *http.Request, key string, recursive b
 		plog.Warningf("auth: no such user: %s.", username)
 		return false
 	}
-	authAsUser := sec.CheckPassword(user, password)
-	if !authAsUser {
```

Comment:
I am not sure how this fixes things. After this, `tom:tom` works? but `tom:` not?